src/redux/modules/events.js

```diff
@@ -29,17 +29,22 @@ export const eventsRequestError = (error) => ({
 });
 
 // potentialIds (payload: { deviceid: String, data: String }) => String[]
-const potentialIds = ({ deviceid = '', data = '' }) => [
-  data.toLowerCase(),
-  data.toUpperCase(),
-  data.toLowerCase().replace(/^3000/, ''),
-  data.toUpperCase().replace(/^3000/, ''),
-  deviceid.toLowerCase(),
-  deviceid.toUpperCase()
-];
+export const potentialIds = ({ deviceid, data }) => {
+  // these now come through as `null` sometimes :S
+  deviceid = typeof deviceid === 'string' ? deviceid : '' + deviceid;
```

Comment:
so the return in the case of `null` or `undefined` will be the strings "null" or "undefined"... I'm not sure (yet) where this code is used but wouldn't it be better to return the types `null` or `undefined` ?